elastic/kibana
# [Migrations] Add `migrator` node role

## Summary

Adds the `migrator` special role to the node roles config:


```yml
#            👇🏻 new
node.roles: ['migrator']
#             or
node.roles: ['background_tasks', 'ui']
#             or 
node.roles: ['*'] # this one is slightly weird now because it actually excludes 'migrator' so it is not truly "all roles", but "all combinable roles"...
```

## How to test

Start Kibana locally and add `node.roles: ['migrator']` to the `kibana.dev.yml`. Kibana should start normally and log:

```
[2023-02-23T12:08:54.123+01:00][INFO ][node] Kibana process configured with roles: [migrator]
```

Note: this role currently does not do anything. This PR just adds the ability to configure it.

Partially addresses https://github.com/elastic/kibana/issues/150295

## Slight improvement to error messages

When specifying known, accepted values but combining with either `migrator` or `*` you will get a message like:

```
[config validation of [node].roles]: wildcard ("*") cannot be used with other roles or specified more than once
```
